### PR TITLE
[ONNX] Update LayerNorm symbolic to handle autocasting

### PR DESCRIPTION
**Pull request by shubhambhokare1**
When onnx export creates a 0-dim tensor of constant type, this action overrides the type promotion logic as quoted in https://github.com/pytorch/pytorch/issues/9515. In order to prevent this from happening this PR adds the following functionality.
If the data type is a floating point type, it is converted to a 0-dim double tensor, else it is converted to a 0-dim tensor of its original type

### CODE REVIEW

**Comment by KsenijaS** [comment]
can you added a description? Is it possible to test this change?

File: torch/onnx/symbolic_opset9.py [hunk 1]
```diff
@@ -1323,6 +1323,9 @@ def layer_norm(g, input, normalized_shape, weight, bias, eps, cudnn_enable):
     eps_cst = g.op("Constant", value_t=torch.tensor(eps))
 
     mean = g.op("ReduceMean", input, axes_i=axes)
+    if mean.type().scalarType() == 'Half':
+        two_cst = g.op("Constant", value_t=torch.tensor(2))
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:1327]: can you try create these two in a way for them to be recognized in scalar_type_analysis pass as a scalar?
And then even better create a helper function for scalar creation.

Comment (BowenBao) [reply torch/onnx/symbolic_opset9.py:1327]: if they are recognized as scalar in that pass, their dtype will have lower priority, and will be overwritten with say half in this case. This way it's more consistent with pt casting logic.

**Review by BowenBao — APPROVED** [review]
